docs/docs/en/user-guide/adapters/fastapi_nats/index.md

```diff
@@ -0,0 +1,209 @@
+# FastAPI + Nats.io
+
+The **FastAPI + Nats.io** Adapter in FastAgency offers the most scalable setup by combining the power of the [**FastAPI**](https://fastapi.tiangolo.com/){target="_blank"} framework for building and exposing workflows as **REST APIs** with the [**Nats.io MQ**](https://nats.io/){target="_blank"} message broker for **scalable and asynchronous** communication. This setup is the **preferred way** for running large workloads in production.
+
+## Use Cases
+
+This section outlines the scenarios where it's particularly beneficial to use the **FastAPI + Nats** Adapter.
+
+### When to Use the FastAPI + Nats.io Adapter
+
+- **Custom Client Applications**: If you want to build your **own client applications** that interact with your FastAgency workflows using REST APIs, this adapter provides greater flexibility and control over the client-side implementation.
+- **High User Demand**: When your application needs to handle a large number of users or messages and requires high scalability, the FastAPI + Nats Adapter is an excellent choice. It is well-suited for building **scalable custom chat applications for larger companies or external customers**.
+- **Conversation Auditing**: If you need the ability to **audit conversations**, the Nats Adapter provides the necessary infrastructure to enable this feature.
+
+
+## Architecture Overview
+
+The following diagram illustrates the high-level architecture of an application using the  **FastAPI + Nats Adapter with Mesop UI** as its frontend:
+
+![Mesop FastAPI](../images/mesop_fastapi_nats.png)
+
+The system is composed of three main components:
+
+### FastAgency Mesop App
+
+The FastAgency Mesop App serves as the frontend interface for the system. It includes:
+
+- **Mesop UI**: A user-friendly web interface for users to interact with the workflows. It communicates with the client and the FastAPI Provider.
+- **FastAPI Provider**: This component acts as a bridge, facilitating seamless communication between the Mesop UI and the FastAPI Adapter.
+
+The FastAgency Mesop App manages all client interactions, handling requests and presenting the results back to the user.
+
+### FastAgency FastAPI App
+
+The FastAgency FastAPI App forms the backend of our system and consists of:
+
+- **Nats Provider**: Responsible for connecting to the Nats Provider, receiving workflow initiation messages, and distributing them to the workers for execution.
+- **FastAPI Adapter**: This component receives requests from the FastAPI Provider and interacts with the Nats Provider for workflow initiation and result retrieval.
```

Comment:
I would probably say something like this:

Communicates with NatsProvider, and implents routes and websocket for FastAPI

Reply:
Content is updated accordingly